stan/math/fwd/mat/vectorize/apply_scalar_binary.hpp

```diff
@@ -0,0 +1,151 @@
+#ifndef STAN_MATH_FWD_MAT_VECTORIZE_APPLY_SCALAR_BINARY_HPP
+#define STAN_MATH_FWD_MAT_VECTORIZE_APPLY_SCALAR_BINARY_HPP
+
+#include <stan/math/prim/mat/vectorize/apply_scalar_binary.hpp>
+#include <stan/math/fwd/core/fvar.hpp>
+
+namespace stan {
+
+  namespace math {
+
+    /**
+     * Template specialization for vectorized functions applying to
+     * fvar<FV> and fvar<FV> arguments.
+     *
+     * @tparam F Type of function defining static apply function.
+     * @tparam FV Value and tangent type for for forward-mode
+     * autodiff variable.
+    */
+    template <typename F, typename FV>
+    struct apply_scalar_binary<F, stan::math::fvar<FV>,
+    stan::math::fvar<FV> > {
+      /**
+       * Function return type, which is <code>fvar<FV></code>.
+       */
+      typedef stan::math::fvar<FV> return_t;
+
+      /**
+       * Apply the function specified by F to the specified arguments.
+       *
+       * @param x Argument fvar<FV>iable.
+       * @param y Argument fvar<FV>iable.
+       * @return Function applied to the fvar<FV>iable.
+       */
+      static inline return_t apply(const stan::math::fvar<FV>& x,
+      const stan::math::fvar<FV>& y) {
+        return F::fun(x, y);
+      }
+    };
+
+    /**
+     * Template specialization for vectorized functions applying to
```

Comment:
Why does it need a specialization for forward mode?  Or more specifically, why isn't a templated base case sufficient?